agreement/abstractions.go

```diff
@@ -229,6 +229,15 @@ type KeyManager interface {
 	// valid for the provided votingRound, and were available at
 	// keysRound.
 	VotingKeys(votingRound, keysRound basics.Round) []account.Participation
+
+	// RecordVote sets the LastVote field for the Account.
+	RecordVote(account basics.Address, round basics.Round) error
+
+	// RecordBlockProposal sets the LastBlockProposal field for the Account.
+	RecordBlockProposal(account basics.Address, round basics.Round) error
```

Comment:
The vote and proposal aren't really that far apart from each other. In fact, tracking the proposal-vote is suffice to indicate a block proposals. I'd collapse these two into a single call, and add a bool indicating whether it's a proposal vote or not.

Reply:
I changed this so there's a single  `Record(basics.Address, basics.Round, <participation type>)` function.

Reply:
One problem with this was that I had to import the node package in the agreement code in order to get this new participation type.

To avoid that, I moved the current keymanager implementation into the data/account directory for this new type. The current partkey db code is in data/account, so this isn't unprecedented. I'm trying to find a better way, so this is may change.

Reply:
moving the `simpleKeyManager` to a different package makes it a non-testing code - which isn't really what we want.

I'm not sure that I followed the reasoning for that, as adding the `account.ParticipationAction` should not change any of the dependencies.

Reply:
These were 2 different changes.

> moving the `simpleKeyManager` to a different package makes it a non-testing code - which isn't really what we want.
I didn't realize that. So public packages that are only imported by _test.go files are excluded from the production binaries?


> I'm not sure that I followed the reasoning for that, as adding the `account.ParticipationAction` should not change any of the dependencies.
Previously it was `node.ParticipationAction` defined in `node/participation.go`, so the agreement abstraction would have had to include node.

I'll continue to think about these two items as I implement the interfaces.